### PR TITLE
Delete Activities

### PR DESCRIPTION
**Pull request by ananyeet**
### What is a quick description of the change?

Add button to delete an activity. 

### Is this fixing an issue?

none.

### Are there more details that are relevant?

Create a button that allows a user to delete an activity. 
The user provides a double confirmation before deletion occurs.

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [x] Tests passing
- [x] Coding style (indentation, etc)

Please explain why any are not present, if any.
No tests written for deleteActivity, this code can be tested by running the website and pressing the buttons themselves.

### CODE REVIEW

File: frontend/src/components/ViewActivities/editActivity.js [hunk 1]
```diff
@@ -96,17 +100,33 @@ class EditActivity extends React.Component {
...
   }
 
+  /**
+   * Delete this activity. 
+   * 
+   * @returns true if the activity was successfully deleted.
+   */
+  async deleteActivity() {
+    if (window.confirm(`Are you sure you want to delete ${this.props.activity[DB.ACTIVITIES_TITLE]}?
+This action cannot be undone!`)) {
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:121]: Ahhhhhhhh this is so nice! I am definitely stealing this if that's cool with you.

File: frontend/src/components/ViewActivities/editActivity.js [hunk 2]
```diff
@@ -96,17 +100,33 @@ class EditActivity extends React.Component {
...
+      const a = await db.collection(DB.COLLECTION_TRIPS).doc(this.props.activity.tripId)
+        .collection(DB.COLLECTION_ACTIVITIES).doc(this.props.activity.id)
+        .delete();
+      return a ? true : false;
+    } else {
+      return false;
+    }
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:128]: See comment in PR #65 about void Promise. Same applied for delete().

File: frontend/src/components/ViewActivities/editActivity.js [hunk 3]
```diff
@@ -96,17 +100,33 @@ class EditActivity extends React.Component {
    */
   countriesDropdown(ref, tzref) {
     return (
-      <Form.Control as="select" ref={ref} onChange={tzref}>
-        <option key="-1">No Change</option>
+      <Form.Control as='select' ref={ref} onChange={tzref}>
+        <option key='-1'>No Change</option>
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:104]: What is the purpose of the -1 key? I couldn't find anything in the Form ReactBootstrap documentation.

Comment (ananyeet) [reply frontend/src/components/ViewActivities/editActivity.js:104]: The -1 keeps that "no change" item at the top of the list. It's replaced with something more intuitive when the form autofills with previously known data in #86.

File: frontend/src/components/ViewActivities/editActivity.js [hunk 4]
```diff
@@ -96,17 +100,33 @@ class EditActivity extends React.Component {
...
           );
         })}
       </Form.Control>
     );
   }
 
+  /**
+   * Delete this activity. 
+   * 
+   * @returns true if the activity was successfully deleted.
```
Comment (Hiram-Silvey) [inline frontend/src/components/ViewActivities/editActivity.js:117]: Update JSdoc to use @return with type.

**Pull request merged**